src/common/lib/client/connection.ts

```diff
@@ -59,7 +59,9 @@ class Connection extends EventEmitter {
   }
 
   get recoveryKey(): string | null {
-    Logger.deprecated('Connection.recoveryKey attribute', 'Connection.createRecoveryKey() method');
+    Logger.deprecationWarning(
+      'The `Connection.recoveryKey` attribute has been replaced by the `Connection.createRecoveryKey()` method. Replace your usage of `recoveryKey` with the return value of `createRecoveryKey()`. `recoveryKey` will be removed in a future version.',
+    );
     return this.createRecoveryKey();
```

Comment:
`createRecoveryKey` is public right? According to spec it should be public

Reply:
Yes, it is — why do you ask?

Reply:
Actually, it wasn't before as per https://ably-real-time.slack.com/archives/C8SPU4589/p1707301881553349?thread_ts=1707230865.713579&cid=C8SPU4589